### PR TITLE
doc: update and re-structure documenation

### PR DESCRIPTION
**Pull request by saininav**
Signed-off-by: Naveen Saini <naveen.kumar.saini@intel.com>

### CODE REVIEW

File: docs/getting-started.md [hunk 1]
```diff
@@ -116,57 +262,135 @@ That will let you start applications but you still can't interact with them: goo
...
 
-### Optional GRUB Configuration Variables
+To switch to linux-intel-acrn-sos LTS 5.4 kernel (default), in 'local.conf' replace with below lines:
+```
+PREFERRED_PROVIDER_virtual/kernel = "linux-intel-acrn-sos"
+PREFERRED_VERSION_linux-intel-acrn-sos = "5.4%"
+```
 
-Below variables are used to generate grub.cfg. These variables can be overwrite in your local config.
+To switch to acrn-kernel-sos, in 'local.conf' replace with below lines:
```
Comment (gvancuts) [inline docs/getting-started.md:363]: What is the difference between this kernel and the `linux-intel-acrn-sos`? Perhaps add a note explaining this?

File: docs/getting-started.md [hunk 2]
```diff
@@ -116,57 +262,135 @@ That will let you start applications but you still can't interact with them: goo
...
+To switch to linux-intel-rt-acrn-uos Preempt-RT 5.4 kernel, in 'conf/multiconfig/uos.conf' replace with below line:
+```
+PREFERRED_PROVIDER_virtual/kernel = "linux-intel-rt-acrn-uos"
+PREFERRED_VERSION_linux-intel-rt-acrn-uos = "5.4%"
+```
+
+To switch to acrn-kernel-uos, in 'conf/multiconfig/uos.conf' replace with below line:
+
+```
+PREFERRED_PROVIDER_virtual/kernel = "acrn-kernel-uos"
```
Comment (gvancuts) [inline docs/getting-started.md:387]: Same question than for the sos kernel, what's the difference with the one previously mentionned?

File: docs/getting-started.md [hunk 3]
```diff
@@ -116,57 +262,135 @@ That will let you start applications but you still can't interact with them: goo
...
+```
+
+### GRUB Configuration
+
+Below variables are used to prepare kernel command lines for Pre-launch VMs and Service OS. Based on these variables ACRN specific grub.cfg get generated.
```
Comment (gvancuts) [inline docs/getting-started.md:392]:
```suggestion
Below variables are used to prepare kernel command lines for Pre-launched User VMs and Service VM. Based on these variables an ACRN specific `grub.cfg` gets generated.
```

File: docs/getting-started.md [hunk 4]
```diff
@@ -116,57 +262,135 @@ That will let you start applications but you still can't interact with them: goo
...
+These variables can be overwrite in your local config.
```
Comment (gvancuts) [inline docs/getting-started.md:393]:
```suggestion
These variables can be overwritten in your local config.
```

File: docs/getting-started.md [hunk 5]
```diff
@@ -116,57 +262,135 @@ That will let you start applications but you still can't interact with them: goo
...
 
 * VMFLAGS - list of pre-launched VMs including Service vm
```
Comment (gvancuts) [inline docs/getting-started.md:395]:
```suggestion
* VMFLAGS - list of pre-launched VMs including Service VM
```

File: docs/getting-started.md [hunk 6]
```diff
@@ -195,9 +419,71 @@ For example, using hybrid scenario for nuc7i7dnb:
 * ACRN_EFI_GRUB2_MOD_CFG wic variable (semicolon (;) sperated list)
```
Comment (gvancuts) [inline docs/getting-started.md:419]:
```suggestion
* ACRN_EFI_GRUB2_MOD_CFG wic variable (semicolon (;) separated list)
```

File: docs/getting-started.md [hunk 7]
```diff
@@ -195,9 +419,71 @@ For example, using hybrid scenario for nuc7i7dnb:
...
     to make additional entries in grub.cfg i.e insmod ext3
 
+For More information, Please check [Update Ubuntu GRUB](https://projectacrn.github.io/latest/tutorials/using_hybrid_mode_on_nuc.html?highlight=hybrid%20mode#update-ubuntu-grub)
```
Comment (gvancuts) [inline docs/getting-started.md:422]:
```suggestion
For more information, please check [Update Ubuntu GRUB](https://projectacrn.github.io/latest/tutorials/using_hybrid_mode_on_nuc.html#update-ubuntu-grub)
```

File: docs/getting-started.md [hunk 8]
```diff
@@ -195,9 +419,71 @@ For example, using hybrid scenario for nuc7i7dnb:
...
+and [acrn-bootconf.bbclass](https://github.com/intel/meta-acrn/blob/master-next/classes/acrn-bootconf.bbclass)
+
+
+### Override distro configuration
+
+Due to parsing sequence conflict with meta-intel, weak assingments are not used in acrn-demo-sos and acrn-demo-uos distros. So to override distro configuration in local.conf, override syntex can be used i.e
```
Comment (gvancuts) [inline docs/getting-started.md:428]:
```suggestion
Due to parsing sequence conflict with `meta-intel`, weak assignments are not used in `acrn-demo-sos` and `acrn-demo-uos` distros. So to override distro configuration in `local.conf`, override syntax can be used i.e
```

File: docs/getting-started.md [hunk 9]
```diff
@@ -195,9 +419,71 @@ For example, using hybrid scenario for nuc7i7dnb:
...
+WKS_FILE_pn-acrn-image-base = "image-installer.wks.in"
+
+# build initramsfs to start the installation
+INITRD_IMAGE_LIVE="core-image-minimal-initramfs"
+
+# make sure initramfs and ext4 image are ready before building wic image
+do_image_wic[depends] += "${INITRD_IMAGE_LIVE}:do_image_complete"
+IMAGE_TYPEDEP_wic = "ext4"
+
+# content to be install
```
Comment (gvancuts) [inline docs/getting-started.md:457]:
```suggestion
# content to be installed
```

File: docs/getting-started.md [hunk 10]
```diff
@@ -195,9 +419,71 @@ For example, using hybrid scenario for nuc7i7dnb:
...
+> Currently ACRN WIC Installer is supported only OE-Core(poky) `master`, `hardknott` and `gatesgarth`. Support for `dunfell` is still in progress.
+
+## Tested Hardware
+The following undergo regular basic testing with their respective MACHINE types.
+
+intel-corei7-64:
+    NUC7i7DNH
+
+intel-skylake-64:
+    NUC7i7DNH
```
Comment (gvancuts) [inline docs/getting-started.md:483]: So it's OK to use a SKL machine type for KBL?

File: docs/getting-started.md [hunk 11]
```diff
@@ -195,9 +419,71 @@ For example, using hybrid scenario for nuc7i7dnb:
...
 
-### Things That Break
 
-If a guest kernel sits at `vhm: initializing` and then restarts then I think the problem is that the UOS is trying to boot with the SOS kernel.
+## Contributing
+You are encouraged to follow Github Pull request workflow to share changes and following commit message guidelines are recommended [OE patch guidelines](https://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines)
 
-If `acrn-dm` segfaults on startup then that is usually that it is trying to do GVT-g but the SOS doesn't have `i915.enable_gvt=1` and friends turned on.
+Layer Maintainer: Naveen Saini
```
Comment (gvancuts) [inline docs/getting-started.md:489]: Perhaps add your email address (or Github profile)?

**Review by gvancuts — COMMENTED** [review]
Very good doc!! :-)

File: docs/getting-started.md [hunk 12]
```diff
@@ -1,18 +1,145 @@
...
+## Supported Hardware
+
+ACRN is supported on below Intel platforms:
+* Apollo Lake
+* Kaby Lake
+* Whiskey Lake
+* Tiger Lake
+* Elkhart Lake
+
+About minimum system requirements and limitations, please find more information at [Supported Hardware](https://projectacrn.github.io/latest/reference/hardware.html)
```
[the remark this answers is not in the document]
Comment (saininav) [reply docs/getting-started.md:58]: This doc is for master-next branch, which is always on latest. So I think it is fine to point to latest here.
When I will backport to master, then I will do the necessary changes.

File: docs/getting-started.md (hunk 10 above)
Comment (saininav) [reply docs/getting-started.md:496]: intel-skylake-64 machine should be use for Sky Lake and its successor platforms. So It is OK to use for KBL.
I have added comment in doc too.

File: docs/getting-started.md [hunk 13]
```diff
@@ -116,59 +261,151 @@ That will let you start applications but you still can't interact with them: goo
...
-BBMULTICONFIG_append  = " installer "
+MACHINE = "intel-corei7-64"
 ```
 
-Then this in `conf/multiconfig/installer.conf`:
+### Adding Guests
 
+The multiconfig/package magic works with a `<image>-package.bb` recipe that inherits `container-package`. This puts the wic image and launcher script into a package which can be added as usual.
+
+To add core-image-base.wic image into Service OS (Add coe-image-base-package to your target ACRN image)
```
Comment (gvancuts) [inline docs/getting-started.md:289]:
```suggestion
To add core-image-base.wic image into Service OS (Add core-image-base-package to your target ACRN image)
```

**Review by gvancuts — APPROVED** [review]
Great doc, thanks!

File: docs/getting-started.md [hunk 14]
```diff
@@ -1,18 +1,144 @@
 # Getting Started
 
-Really rough getting started brain dump.
+This guide will show how to setup host machine for Yocto build and then how to build and boot ACRN Hypervisor, Service VM and User VMs on Intel platforms.
```
Comment (fitchbe) [inline docs/getting-started.md:3]: "how to setup host machine" --> "how to set up the host machine" (the verb form is "set up"; the noun form is "setup")

File: docs/getting-started.md [hunk 15]
```diff
@@ -1,18 +1,144 @@
...
+* Kaby Lake
+* Whiskey Lake
+* Tiger Lake
+* Elkhart Lake
+
+About minimum system requirements and limitations, please find more information at [Supported Hardware](https://projectacrn.github.io/latest/reference/hardware.html)
+
+
+## Setup Build Host
+
```
Comment (fitchbe) [inline docs/getting-started.md:62]: "Setup Build Host" --> "Set Up Build Host"

File: docs/getting-started.md [hunk 16]
```diff
@@ -1,18 +1,144 @@
...
+VII. [Tested Hardware](#tested-hardware)
+
+VIII.[Contributing](#contributing)
+
+## Overview
+This layer provides ACRN Hypvervisor integration with Yocto Project.
+
+## Supported Hardware
+
+ACRN is supported on below Intel platforms:
```
Comment (fitchbe) [inline docs/getting-started.md:52]: "on below" --> "on the following"

File: docs/getting-started.md [hunk 17]
```diff
@@ -1,18 +1,144 @@
...
+```
+
+#### Configure Service VM (SOS)
+
+meta-acrn maintains prototype DISTRO configurations for both Servcie VM OS (SOS) and User VM OS (UOS). `acrn-demo-sos` for Service VM OS and `acrn-demo-uos` for User VM OS.
+`local.conf` carries the common configuration, which can be overwritten by individual multiconfigs `conf/multiconfig/xxx.conf`
+For the Service VM OS, we carry configuration in `conf/local.conf` and for User VM OS we carry in `conf/multiconfig/uos.conf`
+
+
+Append below configuration in `conf/local.conf`:
```
Comment (fitchbe) [inline docs/getting-started.md:141]: "Append below" --> "Append the following"

File: docs/getting-started.md [hunk 18]
```diff
@@ -35,67 +161,86 @@ IMAGE_INSTALL_append_pn-acrn-image-sato = " core-image-weston-package"
 PREFERRED_PROVIDER_virtual/kernel = "linux-intel-acrn-sos"
 ```
 
-Then this in `conf/multiconfig/uos.conf`:
+#### Configure Post-launched User VM (UOS)
+
+Add below in `conf/multiconfig/uos.conf`:
```
Comment (fitchbe) [inline docs/getting-started.md:166]: "Add below" --> "Add the following"

File: docs/getting-started.md [hunk 19]
```diff
@@ -116,59 +261,151 @@ That will let you start applications but you still can't interact with them: goo
...
 
 
-### Install onto NUC
 
-To install the image on to NUC, you could burn the .wic image to the target NUC internal storage.
+## Configurations
+
+### MACHINE Configuration
+
+[meta-intel](https://git.yoctoproject.org/git/meta-intel) provides below machine configurations:
```
Comment (fitchbe) [inline docs/getting-started.md:269]: "provides below" --> "provides the following"

File: docs/getting-started.md [hunk 20]
```diff
@@ -116,59 +261,151 @@ That will let you start applications but you still can't interact with them: goo
...
+* intel-corei7-64
 
+* intel-skylake-64
+
+> intel-skylake-64 is 64-bit machine with -march=skylake and avx2 instruction-set set up. For more information, please check [intel-skylake-64.conf](http://git.yoctoproject.org/cgit/cgit.cgi/meta-intel/tree/conf/machine/intel-skylake-64.conf)
+
+> intel-skylake-64 machine must be used for Skylake and successor platforms
+
+
+To configure MACHINE, set below in your `conf/local.conf`
```
Comment (fitchbe) [inline docs/getting-started.md:280]: "set below" --> "set the following"

File: docs/getting-started.md [hunk 21]
```diff
@@ -116,59 +261,151 @@ That will let you start applications but you still can't interact with them: goo
...
+ACRN_RELEASE = "n"
+```
+
+### Kernel Configuration
+
+There are multiple kernel variant available for both SOS and UOS.
+
+#### SOS
+
+To switch to linux-intel-acrn-sos LTS 5.4 kernel (default), in 'local.conf' replace with below lines:
```
Comment (fitchbe) [inline docs/getting-started.md:359]: "replace with below lines" --> "replace with the following lines" (also line 365 and additional occurrences further on)

**Review by fitchbe — APPROVED** [review]

**Pull request merged**